icu4c/source/i18n/unicode/usimplenumberformatter.h

```diff
@@ -175,14 +208,13 @@ usnumf_openForLocaleAndGroupingStrategy(
 /**
  * Formats a number using this SimpleNumberFormatter.
  *
- * The USimpleNumber is adopted and must not be freed after calling this function,
- * even if the function sets an error code. If you use LocalUSimpleNumberPointer,
- * call `.orphan()` when passing it to this function.
+ * The USimpleNumber is cleared after calling this function. It can be re-used via
+ * usnum_setToInt64.
```

Comment:
What happens if the caller doesn't call `unum_setToInt64()`? Do they get an error?  Do they get the previous value formatted again?

Reply:
Never mind; I see from the unit tests that you just return an `ILLEGAL_ARGUMENT_ERROR`, and that seems fine to me.